namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulNamer.scala

```diff
@@ -17,8 +21,16 @@ object ConsulNamer {
     weights: Map[String, Double] = Map.empty,
     stats: StatsReceiver = NullStatsReceiver
   ): Namer = {
-    val lookup = new LookupCache(consulApi, agentApi, setHost, consistency, preferServiceAddress, weights, stats)
-    new TaggedNamer(lookup, prefix)
+    val taggingParser = (path: Path) =>
```

Comment:
Is it possible to make to avoid this refactor in favor of a smaller change?  For instance, to keep the TaggedNamer and UntaggedNamer classes but have them both extend a trait which adds the handlerPrefix and adminHandler implementations.  I think this lets us avoid needing to create and pass this parser function.

Reply:
Indeed it would be possible, but only if the actual "parser" is passed to the `LookupCache.apply` call each time by the corresponding subclass.

This is subjective, but this way the essential difference between the two Namers is actually captured by that function and nothing more.

Even more, the function is passed to build the specific instance of `LookupCache` because it's there that the Path is now parsed, to avoid passing both the parsed Path elements and the whole Path (needed as a cache key) to the cache lookup call.

I'm not totally against the different subclasses, I'm just pointing out things to consider to decide what's the best solution.

Let me know what you think about this

Reply:
I agree with you on all points.  However, my preference is to avoid refactors like this unless
a) The refactor is clearly superior or b) the refactor is inextricably connected to the change at hand.

Otherwise, I'd prefer to keep this PR focused on the new functionality and relegate any non-essential refactors to separate PRs.

Reply:
Cool then, I'll see to it 👍